### PR TITLE
fix(KillNemesis): report killed nemesis based on critical events

### PR DESCRIPTION
**Pull request by fruch**
since `KillNemesis` was reporting any currently running nemesis during teardown as skipped, reporting looks like it passes regardless if the teardown was cause of end of the test, of cause of critical even stopping the test.

this change would report the nemesis as failure if there are any critical events, and otherwise it would be reported as skipped.

### Testing
- [ ] - ~~https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/longevity-5gb-1h-SlaIncreaseSharesDuringLoad-aws/7/~~
- [x] - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/longevity-5gb-1h-CorruptThenRepairMonkey-aws-test/3

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by vponomaryov — APPROVED** [review]
LGTM

**Review by roydahan — APPROVED** [review]

**Comment by fruch** [comment]
@fgelcer I couldn't easily reproduce those cases, to validate it would works as expected 
do you have a specific job that it happens 100% ? i.e. a critical during a nemesis ?

**Comment by roydahan** [comment]
@fruch You can just use any disruptive nemesis and set the c-s thread to CL=ALL.

**Comment by fruch** [comment]
It's proven to be working but the error message doesn't exactly show what I had in mind, just the backtrace

**Comment by fgelcer** [comment]
> @fgelcer I couldn't easily reproduce those cases, to validate it would works as expected do you have a specific job that it happens 100% ? i.e. a critical during a nemesis ?

`longevity-5gb-1h-GrowShrinkClusterNemesis-aws-test` after some 6 hours it always happens

File: test-cases/nemesis/longevity-5gb-1h-CorruptThenRepairMonkey.yaml [hunk 1]
```diff
@@ -2,8 +2,8 @@ test_duration: 90
 
 prepare_write_cmd:  "cassandra-stress write cl=QUORUM n=5048570 -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=80 -pop seq=1..5048570 -col 'n=FIXED(8) size=FIXED(128)' -log interval=5"
 
-stress_cmd: ["cassandra-stress write cl=QUORUM duration=60m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=40 -pop seq=1..5048570 -col 'n=FIXED(8) size=FIXED(128)' -log interval=5",
-             "cassandra-stress read  cl=QUORUM duration=60m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=40 -pop seq=1..5048570 -col 'n=FIXED(8) size=FIXED(128)' -log interval=5"
+stress_cmd: ["cassandra-stress write cl=ALL duration=60m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=40 -pop seq=1..5048570 -col 'n=FIXED(8) size=FIXED(128)' -log interval=5",
```
Comment (fgelcer) [inline test-cases/nemesis/longevity-5gb-1h-CorruptThenRepairMonkey.yaml:5]: need to revert this change

File: test-cases/nemesis/longevity-5gb-1h-CorruptThenRepairMonkey.yaml [hunk 2]
```diff
@@ -2,8 +2,8 @@ test_duration: 90
...
+             "cassandra-stress read  cl=ALL duration=60m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=40 -pop seq=1..5048570 -col 'n=FIXED(8) size=FIXED(128)' -log interval=5"
```
Comment (fgelcer) [inline test-cases/nemesis/longevity-5gb-1h-CorruptThenRepairMonkey.yaml:6]: need to revert this change

**Review by fgelcer — APPROVED** [review]
other than these 2 changes were done to test it, that need to be reverted, LGTM

**Comment by fruch** [comment]
> other than these 2 changes were done to test it, that need to be reverted, LGTM

removed the testing commit,

found one last issue, the report to ES wasn't correct and was reported as success.
fixed it and now it would be reported as failure both in ES and argus

**Review by roydahan — APPROVED** [review]

**Comment by fruch** [comment]
@roydahan I think the KillNemesis is proven stable, and with this fix we can backport to 2023.1, what do you say ?

**Comment by roydahan** [comment]
For now we use it only for the nightly runs on master, so not urgent for 2023.1
Let's give it some time there and see if it's needed.

**Comment by fruch** [comment]
> For now we use it only for the nightly runs on master, so not urgent for 2023.1
> Let's give it some time there and see if it's needed.

It was needed for the sla test cases, but not critical
